cmd/metadata_generated.go

```diff
@@ -0,0 +1,62 @@
+// Code generated by piper's step-generator. DO NOT EDIT.
+
+package cmd
+
+import "github.com/SAP/jenkins-library/pkg/config"
+
+// GetStepMetadata return a map with all the step metadata mapped to their stepName
+func GetAllStepMetadata() map[string]config.StepData {
```

Comment:
should we maybe use pointers `map[string]*config.StepData` instead?

Reply:
the <stepName>Metadata() methods do not return pointers, we have to change the method interface if we want to return a map of pointers here.

Reply:
can't we just do s.th like 

 `"abapAddonAssemblyKitCheckCVs ":            &abapAddonAssemblyKitCheckCVsMetadata(),`

Reply:
then we don't duplicate all the things again ...
or do I miss something here?

Reply:
The above line is not allowed because the `&` tries to make a pointer out of the _function_, not the returned object.
Do you know about side effects of the current implementation? 
I would rather stick with this for now, if we recognize runtime or memory issues, we could consider generating a conversion function for object to pointer type conversion.